Add tests for the Mini Cart Block when it is added on the FSE editor.

gigitux opened the pull request:
This PR adds tests for the Mini Cart Block when it is added on the FSE editor.
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

<!-- Reference any related issues or PRs here -->
Fixes #5727 

### Testing

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. Run all E2E tests or using this command:
``` 
npm run test:e2e-dev tests/e2e/specs/backend/mini-cart.test.js
```

gigitux committed:
Add tests for the Mini Cart Block when it is added on the FSE editor #5727

Add tests for the Mini Cart Block when it is added on the FSE editor

sunyatasattva commented on tests/e2e/specs/backend/mini-cart.test.js:
```diff
@@ -37,6 +49,16 @@ const addBlockToWidgetsArea = async () => {
+	);
+	await insertButton.click();
+};
+
```
Since this utility function is not specific to the `mini-cart` but could be useful for other tests, what about migrating it in the `utils` file and making it generic?

sunyatasattva reviewed (changes requested):
My only other suggestion, would be to extract your selectors in a `SELECTORS` object instead of hard-coding them within your locators.

You can avoid duplication that way and make the code more expressive.

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-gutenberg-products-block into add/5727-add-test-fse-mini-cart

gigitux committed:
not hardcoding selectors

gigitux committed:
fix typo

sunyatasattva reviewed (approved):
I'm ok with this after the changes, thanks for addressing them! But now the E2E tests are failing 😅 🔫

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-gutenberg-products-block into add/5727-add-test-fse-mini-cart

gigitux commented:
> I'm ok with this after the changes, thanks for addressing them! But now the E2E tests are failing 😅 🔫

The E2E tests are failing for #6071. We have the same error in the trunk. Before merging, I will wait until we resolve it to be sure that this PR doesn't add other errors

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-gutenberg-products-block into add/5727-add-test-fse-mini-cart

sunyatasattva commented:
Probably this should have the `skip-changelog` label, right? Also, it would be nice, in the interest of future testers (like release leads), to indicate which files one should run for E2E tests, making it easier for people to test this PR.

I'm re-running failed jobs in the meantime, as the failure seem unrelated to #6071.

gigitux committed:
Merge branch 'trunk' into add/5727-add-test-fse-mini-cart

gigitux commented:
> Probably this should have the `skip-changelog` label, right? Also, it would be nice, in the interest of future testers (like release leads), to indicate which files one should run for E2E tests, making it easier for people to test this PR.

Yes, I added the label! I will update the description too. Thanks for the feedback!

> I'm re-running failed jobs in the meantime, as the failure seem unrelated to #6071.

Lol, it is very strange. If you check the commit [ae98562](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6000/commits/ae985621d6a4477c6ddb1fdcfe80cc859ff9d69f) and [the related steps](https://github.com/woocommerce/woocommerce-gutenberg-products-block/runs/5640924641?check_suite_focus=true) only the tests with the GB installed fail with the same error that reported on #6071.

I just updated the branch, if I continue to have errors, I will investigate on Monday.

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-gutenberg-products-block into add/5727-add-test-fse-mini-cart

gigitux committed:
Merge branch 'trunk' into add/5727-add-test-fse-mini-cart

gigitux commented:
I tried locally and I'm not able to replicate the error :/ 
I'm re-running the failed jobs

gigitux committed:
Merge branch 'add/5727-add-test-fse-mini-cart' of github.com:woocommerce/woocommerce-gutenberg-products-block into add/5727-add-test-fse-mini-cart

gigitux committed:
restore old searchForBlock function

gigitux committed:
add export

gigitux committed:
Revert "restore old searchForBlock function"

This reverts commit 7049ad85bc2752ad33933eec00da9682943ec35a.

gigitux committed:
use searchForBlock from package only for FSE pages

gigitux committed:
add fix for GB

gigitux committed:
Merge branch 'trunk' into add/5727-add-test-fse-mini-cart

gigitux committed:
Merge branch 'trunk' into add/5727-add-test-fse-mini-cart

gigitux committed:
Merge branch 'trunk' into add/5727-add-test-fse-mini-cart
